xwayland/selection/incoming.c

```diff
@@ -61,9 +61,12 @@ static void xwm_write_property(struct wlr_xwm_selection_transfer *transfer,
 	transfer->property_start = 0;
 	transfer->property_reply = reply;
 
-	xwm_data_source_write(transfer->wl_client_fd, WL_EVENT_WRITABLE, transfer);
-
-	if (transfer->property_reply != NULL) {
+	bool wl_client_finished_consuming =
+		!xwm_data_source_write(transfer->wl_client_fd, WL_EVENT_WRITABLE, transfer);
+	if (!wl_client_finished_consuming) {
```

Comment:
The double negation is a little complicated here. At least it happens in two steps :P

Reply:
Agreed, I tried to compensate by being **very** explicit with the naming :)